dcrpm/yum.py

```diff
@@ -31,12 +32,25 @@ def __init__(self):
         # type: () -> None
         self.logger = logging.getLogger()  # type: logging.Logger
         self.status_logger = logging.getLogger("status")  # type: logging.Logger
+        try:
+            self.yum = which(YUM_CMD_NAME)
+        except DcRPMException:
+            try:
+                self.yum = which(DNF_CMD_NAME)
+            except DcRPMException:
+                self.yum = None
+        if self.yum:
+            self.logger.info("Using %s for yum" % self.yum)
+        else:
+            self.logger.error("Cannot find yum!")
 
     def check_stuck(self, dry_run=False):
         # type: (bool) -> bool
+        if not self.yum:
```

Comment:
doesn't seem useful to do tests if we have NEITHER tool...

Reply:
I agree, and this whole thing should be refactored, but right now we're going for minimal scope of change to get something working.